### PR TITLE
feat: v4 to v5 migration for zero trust list

### PR DESCRIPTION
**Pull request by tamas-jozsa**

### CODE REVIEW

File: internal/testhelpers/state.go
Comment (tamas-jozsa) [inline internal/testhelpers/state.go:1]: **Problem**: The state test helper wasn't correctly simulating how the actual migration framework
  works, causing test failures in account_member and dns_record resources.

  **Root Cause**: The real migration handler (internal/handlers/state_transform.go:83) passes
  individual instance objects to TransformState(), not full state documents. However, the test
  helper was incorrectly passing the entire state structure.

  **Solution**: Updated internal/testhelpers/state.go to:
  1. Detect and handle instance-only test inputs (no "resources" field) by passing them directly
  to TransformState()
  2. For full state tests, iterate through instances and pass each one individually to
  TransformState() - matching real behavior
  3. Properly preserve empty arrays and edge cases like empty resources

  **Impact**: This ensures unit tests accurately validate migration behavior, preventing bugs where
  migrations work in tests but fail in production.

**Review by SirCortly — APPROVED** [review]

File: internal/resources/zero_trust_list/v4_to_v5.go [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+func (m *V4ToV5Migrator) CanHandle(resourceType string) bool {
+	return resourceType == m.oldType
+}
+
+// Preprocess handles the complex transformation of items and items_with_description
+func (m *V4ToV5Migrator) Preprocess(content string) string {
+	// Process each resource block separately to avoid mixing items
+	// Match resource blocks properly including nested blocks
+	resourcePattern := regexp.MustCompile(`(?ms)(resource\s+"cloudflare_(?:teams|zero_trust)_list"\s+"[^"]+"\s*\{(?:[^{}]|\{[^}]*\})*\})`)
```
Comment (vaishakdinesh) [inline internal/resources/zero_trust_list/v4_to_v5.go]: Do we really need to use regex? What exactly is this doing?

File: internal/resources/zero_trust_list/v4_to_v5.go [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+				}
+				currentItems = append(currentItems, item)
+				return true
+			})
+			
+			result, _ = sjson.Set(result, "attributes.items", currentItems)
+			result, _ = sjson.Delete(result, "attributes.items_with_description")
+		}
+		
+		// Don't add computed fields during migration - they should be preserved as-is or removed
```
Comment (vaishakdinesh) [inline internal/resources/zero_trust_list/v4_to_v5.go]: I think this comment isn't really needed.

File: internal/resources/zero_trust_list/v4_to_v5.go (hunk 1 above)
Comment (tamas-jozsa) [reply internal/resources/zero_trust_list/v4_to_v5.go]: we need to process each resource independently, and this regex selects each resource block, so we can do the necessary transformations

**Review by kamarjago — APPROVED** [review]
